perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2984,120 +3183,919 @@ private void closeExtSourcesAfterSynchronization(ExtSource membersSource, ExtSou
+	 *		3.2 There isn't subGroup with name equals to candidateGroup parent name -> specify baseGroup as new parent
+	 * 4. In case parent group hasn't changed, method will return null
+	 *
+	 * Method is used by group structure synchronization.
+	 *
+	 * @param groupToUpdate for which will be parent specified
+	 * @param baseGroup under which may be new parent
+	 * @param candidateGroup with potential new parent group
+	 * @return updated parent group (if null, parent group hasn't changed)
+	 * @throws InternalErrorException
+	 */
+	private Group specifyParentForUpdatedGroup(PerunSession sess, Group groupToUpdate, Group baseGroup, CandidateGroup candidateGroup) throws InternalErrorException {
+
+		String actualParentName;
+		try {
+			actualParentName = getGroupById(sess, groupToUpdate.getParentGroupId()).getShortName();
+		} catch (GroupNotExistsException e) {
+			throw new InternalErrorException(e);
+		}
+
+		Group newParentGroup = null;
+
+		if(candidateGroup.getParentGroupName() == null){
+			if(!actualParentName.equals(baseGroup.getShortName())){
+				newParentGroup = baseGroup;
+			}
+		}
+
+		else if(!actualParentName.equals(candidateGroup.getParentGroupName())){
+			if(!actualParentName.equals(baseGroup.getShortName())) {
+				newParentGroup = baseGroup;
+			}
+			List<Group> subGroups = getAllSubGroups(sess, baseGroup);
+			for (Group subGroup : subGroups) {
+				if (subGroup.getShortName().equals(candidateGroup.getParentGroupName())) {
+					newParentGroup = subGroup;
+					break;
+				}
+			}
+		}
+		return newParentGroup;
+	}
+
+	/**
+	 * Update group description.
+	 *
+	 *  1. Compare descriptions.
+	 * 		1.1 If they differ, set new description into the old one, update whole group and return true.
+	 * 		    Otherwise return false.
+	 *
+	 * Method is used by group structure synchronization.
+	 *
+	 * @param sess
+	 * @param groupWithOldDescription group with description which will be chaged
+	 * @param groupWithNewDescription candidate group with description which will replace the old one
+	 * @return true if description changed, false otherwise
+	 * @throws InternalErrorException
+	 */
+	private boolean updateGroupDescription(PerunSession sess, Group groupWithOldDescription, CandidateGroup groupWithNewDescription) throws InternalErrorException {
+		if(groupWithOldDescription.getDescription() != null) {
+			if(!groupWithOldDescription.getDescription().equals(groupWithNewDescription.asGroup().getDescription())){
+				groupWithOldDescription.setDescription(groupWithNewDescription.asGroup().getDescription());
+				groupsManagerImpl.updateGroup(sess, groupWithOldDescription);
+				return true;
+			}
+		} else if(groupWithNewDescription.asGroup().getDescription() != null){
+			groupWithOldDescription.setDescription(groupWithNewDescription.asGroup().getDescription());
+			groupsManagerImpl.updateGroup(sess, groupWithOldDescription);
+			return true;
+		}
+		return false;
```

Comment:
You are using IMPL version of method updateGroup, where is missing auditer log information about this change!